docs/rfcs/blocks-in-widget-areas.md

```diff
@@ -0,0 +1,535 @@
+```
+PUT /wp/v2/widget-areas/footer
+
+{ "content": { "raw": "
+	<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->
+	<!-- wp:legacy-widget {\"identifier\":\"search-2\",\"instance\":{\"title\":\"\"}} /-->
+	<!-- wp:legacy-widget {\"identifier\":\"recent-posts-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+	<!-- wp:legacy-widget {\"identifier\":\"recent-comments-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+" } }
+```
+
+```json
+{
+	"id": "footer",
+	...
+	"content": {
+		"raw": "
+			<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->
+			<!-- wp:legacy-widget {\"identifier\":\"search-2\",\"instance\":{\"title\":\"\"}} /-->
+			<!-- wp:legacy-widget {\"identifier\":\"recent-posts-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+			<!-- wp:legacy-widget {\"identifier\":\"recent-comments-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+		",
+		"rendered": ...
+	}
+}
+```
+
+This triggers a migration. `'footer'` now points to a numeric ID instead of an array. Other widget areas are not affected.
+
+```php
+array(
+	'footer' => 123,
+	'sidebar' => array(
+		'recent-posts-2',
+		'recent-comments-2',
+	),
+	'array_version' => 4,
+)
+```
+
+Here, `123` is the ID of a `wp_area` post. The `post_content` field of `get_post( 123 )` contains the `'footer'` widget area's block markup.
+
+```html
+<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->
+<!-- wp:legacy-widget {"identifier":"search-2","instance":{"title":""}} /-->
+<!-- wp:legacy-widget {"identifier":"recent-posts-2","instance":{"title":"","number":5}} /-->
+<!-- wp:legacy-widget {"identifier":"recent-comments-2","instance":{"title":"","number":5}} /-->
+```
+
+## New APIs
+
+### `register_widget_area()`
+
+The `register_widget_area()` function is equivalent to [`register_sidebar()`][register_sidebar] except that it does not support `'class'`, `'before_widget'`, `'after_widget'`, `'before_title'` and `'after_title'`.
+
+By encouraging theme developers to use `register_widget_area()` instead of `register_sidebar()`, we encourage themes to allow users to have full control of their widget areas using the block editor.
+
+```php
+function register_widget_area( array|string $attributes = array() )
+```
+
+#### Parameters
+
+- `$attributes`: _(array|string)_ _(optional)_ Arguments for the widget area being registered.
+  - `'name'`: _(string)_ _(optional)_ Name or title of the widget area. Default `"Widget Area $instance"`.
+  - `'id'`: _(string)_ _(optional)_ Unique identifier of the widget area. Default `"widget-area-$instance"`.
+  - `'description'`: _(string)_ _(optional)_ Description the widget area, displayed in the admin interface.
+
+#### Return
+
+_(string)_ ID added to `$wp_registered_sidebars` global.
+
+### `widget_area()`
+
+The `widget_area()` function is an exact alias of the [`dynamic_sidebar()`][dynamic_sidebar] function, provided so that themes can transition away from thinking about "sidebars".
+
+## Backwards Compatibility
+
```

Comment:
Widget Visibility is a highly used piece of functionality in jetpack. This will need to still function after any change. https://jetpack.com/support/widget-visibility/

Reply:
I guess the question here, what API jetpack is using to do this. Feels like something similar to https://wordpress.org/plugins/block-options/

Reply:
For Jetpack, with widgets being posts in a custom post type, they might be able to filter the query to exclude the widgets being loaded on a given route?  In the future this is a feature I think WordPress should offer out of the box.

Reply:
@nerrad The plan is to have one post with lots of blocks, for each widget area. How would that work filtering the query?

Reply:
I clearly misunderstood :).  In that case ya filtering could be harder and wouldn't be filtered by query but would either have to be server side via the response prepared for the api request (i.e. unserializing blocks and using attributes to determine whether the block is returned or not) or client side (via js filtering of the blocks by attribute).

Reply:
Seems like the `render_block` filter would be a good fit for this use-case.

Reply:
What API does Jetpack use to do this?

Reply:
I think they hook deep into filters to remove widgets on conditional logic. It is a little hacky I remember. Someone will need to look into this.

Reply:
I've specified that  `wp_{get,set}_sidebars_widgets()` will be made backwards compatible in ded575402c55ee46a1b4e48a85cdee1cf0c8b733.